src/pages/achievement/control/subcomponents/controlPanelTools/EditableAchievementTask.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+
+import {
+  AchievementItem,
+  FilterStatus
+} from '../../../../../features/achievement/AchievementTypes';
+import AchievementTask from '../../../dashboard/subcomponents/AchievementTask';
+import AchievementInferencer from '../../../dashboard/subcomponents/utils/AchievementInferencer';
+import AchievementControlPanelTools from './AchievementControlPanelTools';
+
+type EditableAchievementTaskProps = {
+  achievement: AchievementItem;
+  inferencer: AchievementInferencer;
+  saveChanges: any;
+};
+
+function EditableAchievementTask(props: EditableAchievementTaskProps) {
+  const { inferencer, achievement, saveChanges } = props;
+
+  const [editableAchievement, setEditableAchievement] = useState<AchievementItem>(achievement);
+  const { id } = editableAchievement;
+
+  const task = (
+    <AchievementTask
+      id={id}
+      inferencer={inferencer}
+      filterStatus={FilterStatus.ALL}
+      displayView={() => {}}
```

Comment:
Maybe use `const noop = ()=>{}` and pass in noop to prevent recreation of function

Reply:
👍